---
title: "third_assignment"
author: "joyfaith onoh"
date: '2022-06-02'
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Q1
## DOWNLOAD THE CODING SEQUENCES OF E.COLI (ecoli)AND CAMPYLOBACTER COLI (ccoli).HOW MANY CODING SEQUENCE ARE PRESENT IN THESE ORGANISMS? HOW MUCH CODING DNA IS THERE IN TOTAL FOR #THESE TWO ORGANISMS? DESCRIBE ANY DIFFERENCES BETWEEN THE TWO ORGANISMS.

```{r}
library("R.utils") # to download the files, you need the chunk below, but because i have already downloaded it, I need not run it again.


#URL="http://ftp.ensemblgenomes.org/pub/bacteria/release-53/fasta/bacteria_0_collection/escherichia_coli_str_k_12_substr_mg1655_gca_000005845/cds/Escherichia_coli_str_k_12_substr_mg1655_gca_000005845.ASM584v2.cds.all.fa.gz"
#download.file(URL,destfile="ecoli_cds.fa.gz")
#gunzip("ecoli_cds.fa.gz")
list.files()
```
```{r}
library("seqinr")
cds <- seqinr::read.fasta("ecoli_cds.fa")
str(head(cds))# head for ecoli
```
```{r}
library("R.utils")
#URL="http://ftp.ensemblgenomes.org/pub/bacteria/release-53/fasta/bacteria_46_collection/campylobacter_coli_gca_003780985/cds/Campylobacter_coli_gca_003780985.PDT000395653.1.cds.all.fa.gz"
#download.file(URL,destfile="ccoli_cds1.fa.gz")
#gunzip("ccoli_cds1.fa.gz")# to unzip the files
cds1 <- seqinr::read.fasta("ccoli_cds1.fa")
str(head(cds1)) #head for ccoli
```
```{r}
#to calculate the number of coding sequence
#cds= ecoli
#cds1= ccoli

length(cds)
length(cds1)
```
#the dna coding sequence of ecoli nis more than double that of ccoli. 


#TOTAL CODING SEQUENCE FOR THE 2 ORGANISMS
```{r}
head(summary(cds))
head(summary(cds1))
```
#Q2
## CALCULATE THE LENGTH OF ALL CODING SEQUENCES IN THESE TWO ORGANISMS. MAKE A BOXPLOT OF CODING SEQUENCE LENGTH IN THESE ORGAMISMS. WHAT IS THE MEAN AND MEDIAN CODING SEQUENCE LENGTH OF THESE TWO ORGANISMS?DESCRIBE ANY DIFFERENCES BETWEEN THE TWO ORGANISMS.
```{r}
len <- sapply(X=cds, FUN= length)#for ecoli
sum(len)

len1 <- sapply(X=cds1, FUN= length)#for ccoli
sum(len1)
```

```{r}
boxplot(len, ylab="LENGTH OF SEQUENCE (bp)", main= "BOXPLOT SHOWING THE LENGTH CODING SEQUENCE IN E.coli",echo=FALSE )

boxplot(len1, ylab= "LENGTH OF SEQUENCE (bp)", main= "BOXPLOT SHOWING THE LENGTH CODING SEQUENCE IN Campylobacter coli",echo=FALSE ) 
```
#TO CALCULATE THE MEAN AND MEDIAN OF THEIR SEQUENCE LENGTH
```{r}
mean(len)
median(len)
mean(len1)
median(len1)
```

QUESTION3
#CALCULATE THE FREQUENCY OF DNA BASES IN THE TOTAL CODING SEQUENCES FOR BOTH ORGANISMS.PERFORM THE SAME CALCULATION FOR THE TOTAL PROTEIN SEQUENCE. CREATE BAR PLOTS FOR NUCLEOTIDE AND AMINO ACID FREQUENCY.DESSCRIBE ANY DIFFERENCES BETWEEN THE TWO ORGANISMS.
```{r}
GC(cds[[1]])
GC(cds[[1]]) 

```
```{r}
count(cds[[1]],1)
count(cds1[[1]],1)
```
```{r}
count(cds[[1]],3)
count(cds1[[1]],3)
```




```{r}
dna <- unlist(cds) # nucleotides for Ecoli
dna1 <- unlist(cds1) #nucleotides for Ccoli
```

```{r,}
GC(dna)
GC(dna1)
```

```{r}
dna_composition <- count(dna,1)
```

barplot(dna_composition,xlab="nucleotides",ylab="frequency", main="E coli CDS composition")
```


```{r}
translate(cds[[1]]) # to calculate the total protein sequence, I first translate
translate(cds1[[1]])# translate for ccoli
```


```{r,}
prot <- lapply(cds,translate) # for Ecoli
prot1 <- lapply(cds1,translate) #for Ccoli
```

#  then define the amino acid alphabet for both organisms
```{r,}
aa <- unique(prot[[2]])# for Ecoli
aa1 <- unique(prot1[[2]])# for Ccoli
```

```{r}
aa <- aa[aa != "*"]
aa1 <- aa1[aa1 != "*"]
length(aa)
length (aa1)
```

```{r}
count(prot[[2]],wordsize=1,alphabet=aa)
count(prot1[[2]],wordsize=1,alphabet=aa1)
```

```{r}
uco(cds[[2]],index="rscu",as.data.frame=TRUE )

uco(cds1[[2]],index="rscu", as.data.frame=TRUE)
```

# IN THE ORGANISM OF INTEREST, IDENTIFY 10 PROTEIN SEQUENCE K-MERS OF LENGTH 3-5 WHICH ARE THE MOST OVERAND UNDER-REPRESENTED K-MERS IN YOUR ORGANISM OF INTEREST. ARE THESE K-MERS ALSO OVER AND UNDER REPRESENTED IN E.COLI TO A SIMILAR EXTENT? PROVIDE PLOTS TOB SUPPORT YOUR OBSERVATIIONS. WHY DO YOU THINK THESE SEQUENCES ARE PRESENT AT DIFFERENT LEVELS IN THE GENOMES OF THESE ORGANISMS?



```{r}
GC(cds[[1]])
GC(cds1[[1]])
```

